extension/encoding/jsonlogencodingextension/extension.go

```diff
@@ -48,20 +71,51 @@ func (e *jsonLogExtension) MarshalLogs(ld plog.Logs) ([]byte, error) {
 			}
 		}
 	}
+
+	// check for processing mode so we can return the best format
+	if !e.config.ArrayMode {
+		var buf bytes.Buffer
+		for i, log := range logs {
+			m, err := json.Marshal(log)
+			if err != nil {
+				return nil, fmt.Errorf("marshaling error with ndjson log: %w", err)
+			}
+
+			buf.Write(m)
+			if i < len(logs)-1 {
+				// if multiple logs, then consider exporting as ndjson
+				buf.WriteByte('\n')
+			}
+		}
+
+		return buf.Bytes(), nil
+	}
+
+	// default mode
```

Comment:
I suggest "array mode" rather than "default mode", since IMO the new mode should become the default :)

Reply:
I think this can go both ways. Personally, I wanted to avoid a breaking change, hence the default fallback on array mode. 

May be would like the opinion of @VihasMakwana on this

Reply:
To be clear, I think it should _eventually_ become the default. Definitely not now. It would need to go through a transition period, if it happens.